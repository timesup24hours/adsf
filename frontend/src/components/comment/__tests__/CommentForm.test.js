import CommentFormWrapper, { CommentForm } from '../CommentForm'
import getMuiTheme from 'material-ui/styles/getMuiTheme';

const date = '2014-04-03T01:40:41.448Z'
const commentObj = {
  _id: '1',
  createdAt: date,
  comment: 'testComment',
}

const comments = [
  commentObj,
  commentObj,
]

const comment = {
  comments: [],
  pending: false,
  success: false,
  fail: false,
  errors: {
    login: 'loginError',
    comment: 'commentError'
  },
  error: false,
  values: {
    comment: ''
  }
}

const commentId = 'comentId'
const commentContent = 'testCommentContent'
const editedComment = 'editedComment'

const store = mockStore({})

test('# CommentFormWrapper', () => {
  const wrapper = shallow(<CommentFormWrapper store={store} />)
  expect(wrapper).toMatchSnapshot()
})

test('# CommentForm', () => {
  const wrapper = shallow(<CommentForm comment={comment} />)
  expect(wrapper).toMatchSnapshot()
})

test('# CommentForm', () => {
  const component = <CommentForm comment={comment} />
  const app = mount(component, {
      context: {
        muiTheme: getMuiTheme(),
      },
      childContextTypes: {
        muiTheme: React.PropTypes.object.isRequired,
      },
    });

  app.find('button').simulate('click');
})
