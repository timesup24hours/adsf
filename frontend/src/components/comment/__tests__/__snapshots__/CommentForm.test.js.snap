exports[`test # CommentForm 1`] = `
<div
  className="CommentForm-container">
  <form
    onSubmit={[Function]}>
    <TextFieldGroupMaterialUI
      className="CommentForm-comment"
      disabled={false}
      error="commentError"
      label="Comment"
      multiLine={true}
      name="comment"
      onChange={[Function]}
      rowsMax={4}
      type="text"
      value="" />
    <div
      className="CommentForm-button">
      <RaisedButton
        disabled={false}
        fullWidth={false}
        label="Comment"
        labelPosition="after"
        primary={false}
        secondary={false}
        type="submit" />
    </div>
  </form>
</div>
`;

exports[`test # CommentFormWrapper 1`] = `
<Component
  commentResetValues={[Function]}
  commentSetError={[Function]}
  commentSetValue={[Function]}
  commentSubmitFail={[Function]}
  commentSubmitRequest={[Function]}
  commentSubmitSuccess={[Function]}
  commentSubmitting={[Function]}
  hideLoading={[Function]}
  showLoading={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  } />
`;
